packages/taquito/src/contract/big-map.ts

```diff
@@ -6,7 +6,12 @@ export class BigMapAbstraction {
   constructor(private id: BigNumber, private schema: Schema, private provider: ContractProvider) {}
 
   async get(keyToEncode: string) {
-    return this.provider.getBigMapKeyByID(this.id.toString(), keyToEncode, this.schema);
+    try {
+      const id = await this.provider.getBigMapKeyByID(this.id.toString(), keyToEncode, this.schema);
+      return id;
+    } catch (e) {
```

Comment:
This seems to return undefined for every all errors that can be thrown by the `this.provider.getBigMapKeyByID` call.

I am not sure if this the behavior described in the ticket. It seems like the correct behavior would be to only return undefined when the key does not exist which would mean that the specific error would be of type `HTTPErrorResponse` see: https://github.com/ecadlabs/taquito/blob/master/packages/taquito-http-utils/src/taquito-http-utils.ts#L21 with a status code of 404

If we do not get this specific error I believe we should still throw the error to the caller and avoid handling it a this level